extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/PanacheQuery.java

```diff
@@ -147,11 +148,29 @@
      */
     public <T extends Entity> T firstResult();
 
+    /**
+     * Returns the first result of the current page index. This ignores the current page size to fetch
+     * a single result.
+     *
+     * @return if found, an optional containing the entity, else <code>Optional.empty()</code>.
+     * @see #singleResultOptional()
+     */
+    public <T extends Entity> Optional<T> firstResultOptional();
+
     /**
      * Executes this query for the current page and return a single result.
      * 
      * @return the single result (throws if there is not exactly one)
      * @see #firstResult()
      */
     public <T extends Entity> T singleResult();
+
+    /**
+     * Executes this query for the current page and return a single result.
+     *
+     * @return if found, an optional containing the entity, else <code>Optional.empty()</code>. Throws if they are more than one
```

Comment:
Well, did you run out of `@throws` tags? ;)

Reply:
Well, humm ... 
For MongoDB we use a RuntimeException and not a dedicated exception for `singleResult` and `singelResultOptional`.
Maybe we should use a more narrowed exception (we didn't have javax.persistance in the classpath) but as it's a RuntimeException I didn't add an `@throws` tag to document it.

Reply:
Just add an exception type that extends runtime, and let's call it a day ;)

Reply:
I choose to create a `PanacheQueryException` so later we can provides a more complex hierarchy if needed (for example `PanacheQLParseException` when parsing an incorect PanacheQL query and `PanacheQueryResultException` to use with those `singleResult` and `singelResultOptional` method).

I create it inside mongodb-panache but I think it will be bettre to put it inside panache-common to be usable by other modules. WDYT ?

Reply:
Sure.

Reply:
Exception moved to `panache-common`